lib/std/unicode.zig

```diff
@@ -629,3 +629,71 @@ test "utf8ToUtf16LeWithNull" {
         testing.expect(utf16[2] == 0);
     }
 }
+
+pub fn utf8ToUtf16LeStringLiteral(comptime utf8: []const u8) *const [calcUtf16LeLen(utf8) :0] u16 {
+    comptime {
+        const len: usize = calcUtf16LeLen(utf8);
+        var utf16le: [len :0]u16 = [_ :0]u16{0} ** len;
+        const utf16le_len = utf8ToUtf16Le(&utf16le, utf8[0..]) catch |err| @compileError(err);
+        assert(len == utf16le_len);
+        return &utf16le;
```

Comment:
Doesn't this return a pointer to a stack variable which no longer exists after the function exists? Or is it fine because of comptime? Still weird though imho. Is it possible to just return the whole array?

Reply:
Comptime values have infinite lifetime

Reply:
Interesting... Not sure what to think of that yet :D

Reply:
Note the corresponding C code:

```c
const char *foo(void) {
    const char *s = "hello";
    return s;
}
```

Same thing.